bundles/mapping/mapmodule/domain/AbstractLayer.js

```diff
@@ -1199,6 +1212,156 @@ Oskari.clazz.define(
             return this._realtime;
         },
         /**
+         * @method hasTimeseries
+         * @return {Boolean}
+         * Has timeseries data
+         */
+        hasTimeseries: function () {
+            return !!this.getAttributes().times;
+        },
+        /**
+         * @method _advancePlayback
+         * Schedule new timeseries frame if needed
+         */
+        _advancePlayback: function () {
+            var me = this;
+            if (me._scheduleNextTimestep) {
+                var nextStep = me._getNextTimestep(this._stepInterval);
+                if (nextStep) {
+                    var imagesToLoad = me._getTileUrls(nextStep);
+                    var millisToTarget = this._frameInterval - Date.now() + me._lastFrameLoadTime;
+                    var millisTimeout = 5000;
+                    me._cancelBuffering = me._imageBuffer.loadImages(imagesToLoad, millisToTarget, Math.max(millisToTarget, millisTimeout), function (success) {
+                        me._setLayerTimestep(nextStep, true);
+                    });
+                } else {
+                    me._stopTimeseriesPlayback();
+                }
+            }
+            me._scheduleNextTimestep = false;
+        },
+        /**
+         * @method _getTileUrls
+         * Get tile urls for given time
+         * @param {String} time time as ISO-string 
+         */
+        _getTileUrls: function (time) {
+            var OLlayers = this._mapModule.getOLMapLayers(this._id);
+            var urls = [];
+            OLlayers[0].grid.forEach(function (a) {
```

Comment:
References to mapping lib specific structures should be moved to mapmodule or plugins. This probably only works for ol2. There's a request for updating url parameters as well. Maybe that could be used instead.